lib/active_merchant/billing/gateways/commerce_hub.rb

```diff
@@ -113,6 +116,9 @@ def add_transaction_interaction(post, options)
       def add_transaction_details(post, options, action = nil)
         post[:transactionDetails] = {}
         post[:transactionDetails][:captureFlag] = options[:capture_flag] unless options[:capture_flag].nil?
+        post[:transactionDetails][:merchantOrderId] = options[:order_id] if options[:order_id].present? && action == 'sale'
+        post[:transactionDetails][:merchantTransactionId] = options[:order_id] if options[:order_id].present? && action == 'sale'
+
```

Comment:
❓ Why did you add to different fields the same value in the same condition?

💭 Also I think you could do something like this:

```ruby
if options[:order_id].present? && action == 'sale'
    post[:transactionDetails][:merchantOrderId] = options[:order_id]
    post[:transactionDetails][:merchantTransactionId] = options[:order_id]
```

Reply:
you are right 👍